presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -530,6 +542,13 @@ private synchronized void startSplit(PrioritizedSplitRunner split)
 
     private synchronized PrioritizedSplitRunner pollNextSplitWorker()
```

Comment:
I think you should move this change to addNewEntrants, which can multiple pollNextSplitWorker calls. 

I was also thinking what do you think of a binary/indicator-variable counter per TaskExecutor: 1 when it is in low mem mode, and 0 otherwise. And that we can even sum that over a cluster to know what fraction of nodes are presently in this mode. This is different from the summation counter you have right now.

Reply:
> I think you should move this change to addNewEntrants, which can multiple pollNextSplitWorker calls.

That's a good point. Updating code with this suggestion.